examples/quantum_die.py

```diff
@@ -28,47 +28,43 @@ def qubits_needed(n):
     """
     return int(math.ceil(math.log(n, 2)))
 
+def qvm(n):
+    """
+    Connect to a QVM that simulates n sides.
+    """
+    if qubits_needed(n) == 9:
+        return qvm = get_qc('9q-square-qvm')
+    else:
+        return qvm = get_qc('%(qubits)dq-qvm' % {"qubits": qubits_needed(n)})
+
 def die_program(n):
     """
     Generate a quantum program to roll a die of n faces.
     """
-    prog = pq.Program()
+    prog = Program()
     qubits = qubits_needed(n)
+    ro = prog.declare('ro', 'BIT', qubits)
     # Hadamard initialize.
     for q in range(qubits):
         prog.inst(H(q))
     # Measure everything.
     for q in range(qubits):
-        prog.measure(q, [q])
-    return prog
+        prog.measure(q, ro[q])
+    return qvm(n).compile(prog)
```

Comment:
you instantiate a `QuantumComputer` here and in `roll_die`. It's better to have only one so the compilation and running steps work together. ie avoid the case where you compile for a qvm and try to run on the qpu